containers/reverse-proxy/envoy-start

```diff
@@ -19,12 +19,29 @@ else
   exit 1
 fi
 
-echo "$(date -Is) Changing the port values in etc/envoy/envoy.yaml to ${SERVICE_PORT} and ${PORT0}"
+if [ ${HEALTH_CHECK_PORT_INDEX:-0} -gt 0 ]; then
+    echo "$(date -Is) HEALTH_CHECK_PORT_INDEX is positive"
+    PROXY_PORT_COUNT=2
+else
+    PROXY_PORT_COUNT=1
+fi
+
+CONFIG_YAML=/etc/envoy/envoy${PROXY_PORT_COUNT}.yaml
+
+echo "$(date -Is) Substituting variables in configuration template"
 
 # change the SERVICE_PORT and PORT0 values in /etc/envoy/envoy.yaml to respective env variables
-sed -i -e "s/SERVICE_PORT/${SERVICE_PORT}/g" -e "s/PORT0/${PORT0}/g" /etc/envoy/envoy.yaml
+(
+set -x
+sed -i \
+    -e "s/{{APP_PORT}}/${PORT0}/g" \
+    -e "s/{{HC_PORT}}/$(( PORT0 + HEALTH_CHECK_PORT_INDEX ))/g" \
+    -e "s/{{PROXY_APP_PORT}}/${SERVICE_PORT}/g" \
+    -e "s/{{PROXY_HC_PORT}}/$(( SERVICE_PORT + HEALTH_CHECK_PORT_INDEX ))/g" \
```

Comment:
👍

Reply:
Can we use the following names: `PROXY_SERVICE_PORT` and `PROXY_ HEALTH_CHECK_PORT`?

Reply:
Renamed all four variables used in the templates:
```
SERVICE_PORT
HEALTH_CHECK_PORT
PROXY_SERVICE_PORT
PROXY_HEALTH_CHECK_PORT
```